include/ddog_profiling_utils.hpp

```diff
@@ -5,12 +5,122 @@
+#include "map_utils.hpp"
+#include "mapinfo_table.hpp"
+#include "symbol.hpp"
+#include "unwind_output.hpp"
 
 #include <string_view>
+#include <vector>
 
+namespace ddprof {
 inline ddog_CharSlice to_CharSlice(std::string_view str) {
   return {.ptr = str.data(), .len = str.size()};
 }
+
+inline void write_function(const Symbol &symbol, ddog_prof_Function *ffi_func) {
+  ffi_func->name = to_CharSlice(symbol._demangle_name);
+  ffi_func->system_name = to_CharSlice(symbol._symname);
+  ffi_func->filename = to_CharSlice(symbol._srcpath);
+  // Not filed (can be computed if needed using the start range from elf)
+  ffi_func->start_line = 0;
+}
+
+inline void write_function(std::string_view demangle_name,
+                           std::string_view file_name,
+                           ddog_prof_Function *ffi_func) {
+  ffi_func->name = to_CharSlice(demangle_name);
+  ffi_func->system_name = {.ptr = nullptr, .len = 0};
+  ffi_func->filename = to_CharSlice(file_name);
+  // Not filed (can be computed if needed using the start range from elf)
+  ffi_func->start_line = 0;
+}
+
+inline void write_mapping(const MapInfo &mapinfo,
+                          ddog_prof_Mapping *ffi_mapping) {
+  ffi_mapping->memory_start = mapinfo._low_addr;
+  ffi_mapping->memory_limit = mapinfo._high_addr;
+  ffi_mapping->file_offset = mapinfo._offset;
+  ffi_mapping->filename = to_CharSlice(mapinfo._sopath);
+  ffi_mapping->build_id = to_CharSlice(mapinfo._build_id);
+}
+
+inline void write_location(const FunLoc *loc, const MapInfo &mapinfo,
+                           const Symbol &symbol,
+                           ddog_prof_Location *ffi_location,
+                           bool use_process_adresses) {
+  write_mapping(mapinfo, &ffi_location->mapping);
+  write_function(symbol, &ffi_location->function);
+  ffi_location->address = use_process_adresses ? loc->ip : loc->_elf_addr;
+  ffi_location->line = symbol._lineno;
+}
+
+// demangling caching based on stability of unordered map
+// This will be moved to the backend
+inline std::string_view get_or_insert_demangled_sym(
+    const char *sym,
+    ddprof::HeterogeneousLookupStringMap<std::string> &demangled_names) {
+  auto it = demangled_names.find(sym);
+  if (it == demangled_names.end()) {
+    const std::string demangled_name =
+        ddprof::Demangler::non_microsoft_demangle(sym);
+    it = demangled_names.insert({std::string(sym), demangled_name}).first;
+  }
+  return it->second;
+}
+
+inline DDRes write_location_blaze(
+    ProcessAddress_t ip,
+    ddprof::HeterogeneousLookupStringMap<std::string> &demangled_names,
+    const MapInfo &mapinfo, const blaze_sym *blaze_sym, unsigned &cur_loc,
+    std::span<ddog_prof_Location> locations_buff) {
+  if (cur_loc >= locations_buff.size()) {
+    return ddres_warn(DD_WHAT_UW_MAX_DEPTH);
+  }
+  constexpr std::string_view undef{};
+  constexpr std::string_view undef_inlined = undef;
+  for (unsigned i = 0; i < blaze_sym->inlined_cnt && cur_loc < kMaxStackDepth;
```

Comment:
The Inlining can not be triggered. As I did not want to create too many changes to measure the operational impact of the PR.
Though we are just one flag away from having it.